lib/multipass_ex/coding.ex

```diff
@@ -0,0 +1,55 @@
+defmodule MultipassEx.Coding do
+  @moduledoc """
+  Encoding / Decoding of Multipass data.
+
+  The MultipassEx gem truncates padding characters from base64 encoding whether
+  they are present or not. It then adds a single padding character to the end
+  of the base64 encoded string before decoding without reguard to its necessity.
+  This causes issues if you decode without `padding: false` as it causes padding
+  errors. Encoding should also be done without padding.
+
+  The gem also replaces characters after base64 encoding to make the string
+  "url safe". This happens before base64 decoding and after base64 encoding.
+  """
+
+  alias MultipassEx.DecodingError
+
+  @doc """
+  Base64 encode the string without padding and replace characters to match the
+  ruby gems "url_safe" flag.
+  """
+  @spec encode(binary) :: binary
```

Comment:
i believe this'll output a `String.t()`?

Reply:
`Base.encode64/1` returns `binary` and then `String.replace` returns `t` where `t` is the first param (`binary`).

Reply:
hm, `t` doesn't indicate a variable, it means the `@type t` from the module.

Reply:
man I totally flashed back to the scala days with that, sorry. You are correct, I'll fix it.